### DISCUSSION TITLE
CMake build errors?

### DISCUSSION THREAD

**Discussion by jonathanc-n** [apache/kvrocks#2523] [Q&A] [answered]
I am running into errors:

Fetching gtest...
CMake Warning (dev) at /opt/homebrew/Cellar/cmake/3.30.3/share/cmake/Modules/FetchContent.cmake:1953 (message):
  Calling FetchContent_Populate(gtest) is deprecated, call
  FetchContent_MakeAvailable(gtest) instead.  Policy CMP0169 can be set to
  OLD to allow FetchContent_Populate(gtest) to be called directly for now,
  but the ability to call it with declared details will be removed completely
  in a future version.
Call Stack (most recent call first):
  cmake/utils.cmake:31 (FetchContent_Populate)
  cmake/gtest.cmake:27 (FetchContent_MakeAvailableWithArgs)
  CMakeLists.txt:151 (include)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Error: Error: generator : Unix Makefiles
Does not match the generator used previously: Ninja
Either remove the CMakeCache.txt file and CMakeFiles directory or choose a different binary directory.

Do you guys get this same error when running './x.py build'?

**Comment by PragmaTwice** [comment, marked as answer]
```
CMake Error: Error: generator : Unix Makefiles
Does not match the generator used previously: Ninja
Either remove the CMakeCache.txt file and CMakeFiles directory or choose a different binary directory.
```

As the error states, you need to build in a new directory if the generator is changed.
Or you can delete the current build dir.

**Reply by jonathanc-n** [reply]
Oh icic, I didn't know to delete the current build. Thank you!
